kaggle_cli/submit.py

```diff
@@ -37,10 +36,7 @@ def take_action(self, parsed_args):
         competition = config.get('competition', '')
         zip_flag = config.get('zip', 'no')
 
-        if Submit._str2bool(zip_flag):
-            zip = True
-        else:
-            zip = False
+        zip = Submit._str2bool(zip_flag)
```

Comment:
Overriding the built-in function name is discouraged. You can just apply the `_str2bool` to the `zip_flag = config.get('zip', 'no')` , i.e. `zip_flag = Submit._str2bool(config.get('zip', 'no'))`.

But I'll address the boolean coercion issue at config provider level, you can get a boolean by `config.get('zip')` after I updated it.

Reply:
The type coercion enhancement has been done, see https://github.com/floydwch/kaggle-cli/commit/9158c36b2a294e685d610afd0dbb0b2009ee532e and https://github.com/floydwch/kaggle-cli/commit/551a1b2d259ef996517f245b1f34b97f55326ca3 .

You can now `config.get('zip', False)`.